ddtrace/llmobs/_integrations/anthropic.py

```diff
@@ -0,0 +1,147 @@
+from ddtrace.llmobs._constants import INPUT_MESSAGES
+from ddtrace.llmobs._constants import METADATA
+from ddtrace.llmobs._constants import MODEL_NAME
+from ddtrace.llmobs._constants import OUTPUT_MESSAGES
+from ddtrace.llmobs._constants import SPAN_KIND
+
+from .base import BaseLLMIntegration
+
+
+log = get_logger(__name__)
+
+
+API_KEY = "anthropic.request.api_key"
+MODEL = "anthropic.request.model"
+TOTAL_COST = "anthropic.tokens.total_cost"
+TYPE = "anthropic.request.type"
+
+
+class AnthropicIntegration(BaseLLMIntegration):
+    _integration_name = "anthropic"
+
+    def llmobs_set_tags(
+        self,
+        resp: Any,
+        span: Span,
+        args: List[Any],
+        kwargs: Dict[str, Any],
+        err: Optional[Any] = None,
+    ) -> None:
+        """Extract prompt/response tags from a completion and set them as temporary "_ml_obs.*" tags."""
+        # if not self.llmobs_enabled:
+        #     return
+        parameters = {
+            "temperature": float(span.get_tag("anthropic.request.parameters.temperature") or 1.0),
+            "max_tokens": int(span.get_tag("anthropic.request.parameters.max_tokens") or 0),
+        }
+        messages = get_argument_value(args, kwargs, 0, "messages")
+        input_messages = self._extract_input_message(messages)
+
+        span.set_tag_str(SPAN_KIND, "llm")
+        span.set_tag_str(MODEL_NAME, span.get_tag("anthropic.request.model") or "")
+        span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages))
+        span.set_tag_str(METADATA, json.dumps(parameters))
+        if err or resp is None:
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps([{"content": ""}]))
+        else:
+            output_messages = self._extract_output_message(resp)
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps(output_messages))
+
+        usage = _get_attr(resp, "usage")
+        self.record_usage(
+            span,
+            {"prompt": _get_attr(usage, "input_tokens", 0), "completion": getattr(usage, "output_tokens", 0)},
+        )
+
+    def _set_base_span_tags(
+        self,
+        span: Span,
+        model: Optional[str] = None,
+        api_key: Optional[str] = None,
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """Set base level tags that should be present on all Anthropic spans (if they are not None)."""
+        if model is not None:
+            span.set_tag_str(MODEL, model)
+        if api_key is not None:
+            if len(api_key) >= 4:
+                span.set_tag_str(API_KEY, f"...{str(api_key[-4:])}")
+            else:
+                span.set_tag_str(API_KEY, api_key)
+
+    def _extract_input_message(self, messages):
+        """Extract input messages from the stored prompt.
+        Anthropic allows for messages and multiple texts in a message, which requires some special casing.
+        """
+        if not isinstance(messages, Iterable):
+            log.warning("Anthropic input must be a list of messages.")
```

Comment:
Do we need to return in this case?

Reply:
In this case I would assume that anthropic will be returning an error, and just logging the warning should be fine.